src/opf/acp.jl

```diff
@@ -1,293 +1,167 @@
+
+    for e in 1:E
+        # Ohm's law
+        model[:ohm_pf][e] = @constraint(model,
+            branch_status[e] * ( gff[e] * wf[e] + gft[e] * wr[e] + bft[e] * wi[e]) - pf[e] == 0
         )
-        model[:ohm_reactive_fr][i] = JuMP.@constraint(model, 
-            - (( b+b_fr)/ttm) * vm_fr^2
-            - ((-b*tr-g*ti)/ttm) * (vm_fr*vm_to*cos(va_fr-va_to))
-            + ((-g*tr+b*ti)/ttm) * (vm_fr*vm_to*sin(va_fr-va_to))
-            - q_fr
-            ==
-            0
+        model[:ohm_qf][e] = @constraint(model,
+            branch_status[e] * (-bff[e] * wf[e] - bft[e] * wr[e] + gft[e] * wi[e]) - qf[e] == 0
         )
-
-        # To side of the branch flow
-        model[:ohm_active_to][i] = JuMP.@constraint(model,
-              (g+g_to) * vm_to^2 
-            + ((-g*tr-b*ti)/ttm) * (vm_to*vm_fr*cos(va_to-va_fr))
-            + ((-b*tr+g*ti)/ttm) * (vm_to*vm_fr*sin(va_to-va_fr)) 
-            - p_to
-            ==
-            0
+        model[:ohm_pt][e] = @constraint(model,
+            branch_status[e] * ( gtt[e] * wt[e] + gtf[e] * wr[e] - btf[e] * wi[e]) - pt[e] == 0
         )
-        model[:ohm_reactive_to][i] = JuMP.@constraint(model,
-            - (b+b_to) * vm_to^2 
-            - ((-b*tr+g*ti)/ttm) * (vm_to*vm_fr*cos(va_to-va_fr)) 
-            + ((-g*tr-b*ti)/ttm) * (vm_to*vm_fr*sin(va_to-va_fr)) 
-            - q_to
-            ==
-            0
+        model[:ohm_qt][e] = @constraint(model,
+            branch_status[e] * (-btt[e] * wt[e] - btf[e] * wr[e] - gtf[e] * wi[e]) - qt[e] == 0
         )
-
+    
+        # Thermal limit
+        model[:sm_fr][e] = @constraint(model, pf[e]^2 + qf[e]^2 ≤ smax[e]^2)
+        model[:sm_to][e] = @constraint(model, pt[e]^2 + qt[e]^2 ≤ smax[e]^2)
+    
         # Voltage angle difference limit
-        model[:voltage_difference_limit][i] = JuMP.@constraint(model, branch["angmin"] <= va_fr - va_to <= branch["angmax"])
-
-        # Apparent power limit, from side and to side
-        model[:thermal_limit_fr][i] = JuMP.@constraint(model, p_fr^2 + q_fr^2 <= branch["rate_a"]^2)
-        model[:thermal_limit_to][i] = JuMP.@constraint(model, p_to^2 + q_to^2 <= branch["rate_a"]^2)
+        model[:va_diff][e] = @constraint(model,
+            dvamin[e] ≤ branch_status[e] * (va[bus_fr[e]] - va[bus_to[e]]) ≤ dvamax[e]
+        )
     end
 
     #
     #   III. Objective
     #
-    # check that we don't have quadratic objective coeffs
-    l, u = extrema(gen["cost"][1] for (i, gen) in ref[:gen])
-    (l == u == 0.0) || @warn "Data $(data["name"]) has quadratic cost terms; those terms are being ignored"
-    JuMP.@objective(model, Min, sum(
-        gen["cost"][2]*pg[i] + gen["cost"][3]
-        for (i,gen) in ref[:gen]
-    ))
-
-    return OPFModel{ACOPF}(data, model)
-end
 
+    @objective(model,
+        Min,
+        sum(c2[g] * pg[g]^2 + c1[g] * pg[g] + c0[g] for g in 1:G if gen_status[g])
```

Comment:
Same comment for quadratic objective as SOC-OPF: I'd rather we support that in a later PR, so we can write the duality part cleanly (I know, less of an issue for AC-OPF than SOC, but I prefer to be consistent across all the formulations)
```suggestion
        sum(c1[g] * pg[g] + c0[g] for g in 1:G if gen_status[g])
```